### PR TITLE
For #22271 Improve URL accessing from the clipboard for Android 12 and above.

### PR DESCRIPTION
**Pull request by Amejia481**
As accessing the clipboard is sensitive action, on Android 12 and above we changed the UI to not show (access the clipboard) the URL unless the user request to do so.

### Behavior on Android 12 and above

We show the **"Fill link from clipboard"** item (but not the URL item)  when the user clicks it, we fill the toolbar with its content.

Demo

### Behavior pre Android 12 and above

Same as before show the **"Fill link from clipboard"** item and the URL item, when the user clicks it, we navigate to the site.

clipboard_pre_android_12

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Review by csadilek — COMMENTED** [review]
👍 I think this is a good immediate solution without losing any functionality, and we can improve later. I noticed two things though

- After filling in from clipboard the cursor is at the beginning instead of at the end. After pasting text the user expectation is for the cursor to be at the end of the text.
- The keyboard pops up and down when filling in.

File: app/src/main/java/org/mozilla/fenix/utils/ClipboardHandler.kt
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/utils/ClipboardHandler.kt]: @csadilek  I added the `try ` as I noticed that `getConfidenceScore` was throwing an `IllegalStateException` when the URL was too long.

Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/utils/ClipboardHandler.kt]: > 👍 I think this is a good immediate solution without losing any functionality, and we can improve later. I noticed two things though
> 
>     * After filling in from clipboard the cursor is at the beginning instead of at the end. After pasting text the user expectation is for the cursor to be at the end of the text.
> 
>     * The keyboard pops up and down when filling in.

Added [this](https://github.com/mozilla-mobile/fenix/pull/22391/commits/27c153937b93fa0a059ff73b26adcc5b275925cc#) separated commit to address the listed issues

**Review by csadilek — APPROVED** [review]
👍

File: app/src/main/java/org/mozilla/fenix/utils/ClipboardHandler.kt
Comment (csadilek) [reply app/src/main/java/org/mozilla/fenix/utils/ClipboardHandler.kt]: Tested again, works great now!

**Comment by Amejia481** [comment]
We will wait until early next week to land, after the long weekend pass.

**Comment by Amejia481** [comment]
@Mergifyio backport releases_v95.0.0

**Pull request merged**
